app/services/app/app.ts

```diff
@@ -160,6 +162,20 @@ export class AppService extends StatefulService<IAppState>
     this.FINISH_LOADING();
   }
 
+  /**
+   * Downloads and installs an overlay
+   * @param url the URL of the overlay
+   */
+  async installOverlay(url: string, progressCallback?: (info: IDownloadProgress) => void) {
+    this.START_LOADING();
+
+    const pathName = await this.overlaysPersistenceService.downloadOverlay(url, progressCallback);
```

Comment:
looks like we catch infinity load if `downloadOverlay` fails. Or I miss something?

Reply:
Yeah, should probably `try/catch` and finish loading then re-raise